ucp/_libs/send_recv.pyx

```diff
@@ -82,43 +83,47 @@ def tag_send(uintptr_t ucp_ep, buffer, size_t nbytes,
     cdef ucp_ep_h ep = <ucp_ep_h><uintptr_t>ucp_ep
     cdef void *data = <void*><uintptr_t>(get_buffer_data(buffer,
                                          check_writable=False))
-    cdef ucp_send_callback_t _send_cb = <ucp_send_callback_t>_send_callback
-    cdef ucs_status_ptr_t status = ucp_tag_send_nb(ep,
-                                                   data,
-                                                   nbytes,
-                                                   ucp_dt_make_contig(1),
-                                                   tag,
-                                                   _send_cb)
+    cdef ucp_send_callback_t _send_cb
+    cdef ucs_status_ptr_t status
+    with nogil:
```

Comment:
Should this be with the gil since it's a callback executed by C ?

Reply:
Our Cython callbacks are acquiring the GIL before they call into Python. So we need to be sure to release the GIL beforehand.